### ISSUE TITLE
compiler-llvm error, "PHINode should have one entry for each predecessor of its parent basic block!"

### ISSUE THREAD

**Issue by nlewycky** [wasmerio/wasmer#2235]
Testcase:

```
(module
  (func (;0;) (result i64 f32)
    i64.const 0
    loop (param i64) (result i64 f32)  ;; label = @1
      i64.popcnt
      i64.extend32_s
      i32.const 0
      br_table 0 (;@1;) 0 (;@1;)
      unreachable
    end))
```

crashes with LLVM verifier:

```
$ wasmer --llvm crash.wat --enable-verifier
PHINode should have one entry for each predecessor of its parent basic block!
  %5 = phi i64 [ 0, %start_of_code ], [ %8, %loop_body ]
PHINode should have one entry for each predecessor of its parent basic block!
  %9 = phi i64 [ %8, %loop_body ]
in function f0
LLVM ERROR: Broken function found, compilation aborted!
Aborted
```

**Comment by ptitSeb** [comment]
It's still happening, but what exact result is expected here?

**Comment by syrusakbary** [comment]
The exact result is that the LLVM IR verification passes. That means probably inspecting the LLVM IR and seeing what is the issue.

Basically, this should not break `wasmer --llvm crash.wat --enable-verifier`

**Comment by ptitSeb** [comment]
Extract of generated IR:
```
entry:
  br label %start_of_code

start_of_code:                                    ; preds = %entry
  br label %loop_body

return:                                           ; preds = %loop_outer
  %1 = phi i64 [ %9, %loop_outer ]
  %2 = phi float [ 0.000000e+00, %loop_outer ]
  %3 = insertvalue { i64, float } undef, i64 %1, 0
  %4 = insertvalue { i64, float } %3, float %2, 1
  ret { i64, float } %4

loop_body:                                        ; preds = %loop_body, %loop_body, %start_of_code
  %5 = phi i64 [ 0, %start_of_code ], [ %8, %loop_body ]
  %6 = call i64 @llvm.ctpop.i64(i64 %5)
  %7 = trunc i64 %6 to i32
  %8 = sext i32 %7 to i64
  switch i32 0, label %loop_body [
    i32 0, label %loop_body
  ]

loop_outer:                                       ; No predecessors!
  %9 = phi i64 [ %8, %loop_body ]
  br label %return
```

**Comment by syrusakbary** [comment]
This was fixed by #2673. This issue no longer exists in master
